apps/src/gamelab/GameLabSprite.js

```diff
@@ -467,8 +468,10 @@ var AABBops = function (p5Inst, type, target, callback) {
                 var newVel1 = dx1.mult(m1 * p5.Vector.sub(this.velocity, other.velocity).dot(dx1) / magnitude);
                 var newVel2 = dx2.mult(m2 * p5.Vector.sub(other.velocity, this.velocity).dot(dx2) / magnitude);
```

Comment:
I think you can also pull the calculations here of `newVel1` and `newVel2` into your `if (modifyPosition) {` condition.